features/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/extensions/assets/site/pages/assets.jag

```diff
@@ -207,16 +207,53 @@ require('/modules/store.js').exec(function (ctx) {
         options.lifecycleState = ["published"];
         assets = store.taggeds(type, options, paging);
     } else if (query) {
-        var urlQuery = 'query=' + query + '&';
-        query = require('/modules/search.js').build(query);
-        query.overview_treatAsASite = 'TRUE';
-        if (!isMultipleVersionDisplayEnabled) {
-            query.overview_makeAsDefaultVersion = 'true';
+        var searchParameter;
+        var searchValue;
+        var splitQuery = query.split(":");
+        searchParameter = splitQuery[0];
+        // This is a registry based search. In search box, user wants to search by business owner name. But
+        // in registry, there is only business owner Id. Therefore, it is necessary to get business owner Ids
+        // related to search prefix from DB.
+
+        if (!searchParameter.equals("Business Owner")) {
+            var urlQuery = 'query=' + query + '&';
+            query = require('/modules/search.js').build(query);
+            query.overview_treatAsASite = 'TRUE';
+            if (!isMultipleVersionDisplayEnabled) {
+                query.overview_makeAsDefaultVersion = 'true';
+            }
+            options = {
+                type: type,
+                attributes: query
+            };
+        } else {
+            searchValue = splitQuery[1];
+            var initialQuery = "businessOwner:" + searchValue;
+            var searchPrefix = searchValue.split("\"")[1];
+            var resultIdsList = storeHostObj.getBusinessOwnerIdsByBusinessOwnerNameField(searchPrefix, tenantId);
+            if (resultIdsList){
+                var resultAssets = [];
+                resultIdsList = resultIdsList.toArray();
+                for(var key in resultIdsList) {
```

Comment:
Formatting issues are fixed in https://github.com/wso2/carbon-appmgt/pull/539/files